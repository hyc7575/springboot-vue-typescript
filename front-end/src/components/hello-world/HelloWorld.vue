<template>
	<div class="hello">
		<h1>{{ msg }}</h1>
		<img src="/assets/logo.png">
		<button @click="clickHandler">
			button
		</button>
	</div>
</template>

<script lang="ts">
import Vue from "vue";
import Component from "vue-class-component";

interface HelloWorldInterface {
	msg: string;
	clickHandler(): void;
}

@Component({
	created() {
		console.log('HelloWorld created');
	}
})
export default class HelloWorld extends Vue implements HelloWorldInterface {
	msg = "Hello!!";
	clickHandler() {
		console.log('click Handler');
	}
}
</script>
