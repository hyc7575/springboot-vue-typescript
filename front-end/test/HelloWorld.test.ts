import { mount } from 'vue-test-utils';
import { createRenderer } from 'vue-server-renderer';
import HelloWorld from "../src/components/hello-world/HelloWorld.vue";

describe('HelloWorld.vue', () => {
    const renderer = createRenderer();
    it('should initialize correctly', () => {
        const wrapper = mount(HelloWorld);
        
        renderer.renderToString(wrapper.vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    })
});
