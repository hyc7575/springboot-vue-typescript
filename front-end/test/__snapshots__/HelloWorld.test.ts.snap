// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue should initialize correctly 1`] = `
<div class="hello">
    <h1>Hello!!</h1>
    <img src="/assets/logo.png">
    <button>
        button
    </button>
</div>
`;
